bokeh/models/formatters.py

```diff
@@ -200,6 +205,50 @@ class CategoricalTickFormatter(TickFormatter):
     """
     pass
 
+class FuncTickFormatter(TickFormatter):
+    """ Format a function based on a python function that is transpiled to
+    javascript via PyScript
+    """
+
+    @classmethod
+    def from_py_func(cls, func):
+        """ Create a CustomJS instance from a Python function. The
+        function is translated to Python using PyScript.
+        """
+        if not isinstance(func, FunctionType):
+            raise ValueError('CustomJS.from_py_func needs function object.')
+        pyscript = import_required('flexx.pyscript',
+                                   'To use Python functions for CustomJS, you need Flexx ' +
+                                   '("conda install -c bokeh flexx" or "pip install flexx")')
+
+        arg = inspect.getargspec(func)[0]
+        if len(arg) != 1:
+            raise ValueError("Can only have one argument. You have: %d" % len(arg))
+
+        # Set the transpiled functions as `formatter` so that we can call it
+        code = pyscript.py2js(func, 'formatter')
+        # We wrap the transpiled function into an anonymous function with a single
+        # arg that matches that of func.
+        wrapped_code = "function (%s) {%sreturn formatter(%s)};" % (arg[0], code, arg[0])
+
+        return cls(code=wrapped_code, lang='javascript')
+
+    code = String(default="", help="""
+    An anonymous JavaScript/CoffeeScript function to reformat a single tick to
+    the desired format.
+
+    .. warning::
+        The anonymous can have only a single argument
```

Comment:
Probably better to just replace "anonymous" with "function" here.   Saying "anonymous" in the first sentence seems ok to me, though.